src/stdx/bounded_array.zig

```diff
@@ -0,0 +1,68 @@
+const std = @import("std");
+const assert = std.debug.assert;
+
+/// A version of `std.BoundedArray` with TigerBeetle-idiomatic APIs.
+pub fn BoundedArray(comptime T: type, comptime capacity: usize) type {
+    const Inner = @import("std").BoundedArray(T, capacity); // smuggle the std version past tidy
+
+    return struct {
+        inner: Inner = Inner{},
+
+        const Self = @This();
+
+        pub inline fn from_slice(items: []const T) error{Overflow}!Self {
+            return .{ .inner = try Inner.fromSlice(items) };
+        }
+
+        pub inline fn count(array: *const Self) usize {
+            return array.inner.len;
+        }
+
+        /// Returns count of elements in this BoundedArray in the specified integer types,
+        /// checking at compile time that it indeed can represent the length.
+        pub inline fn count_as(array: *const Self, comptime Int: type) Int {
+            return array.inner.len;
+        }
+
+        pub inline fn full(self: Self) bool {
+            return self.count() == capacity;
+        }
+
+        pub inline fn empty(self: Self) bool {
```

Comment:
I've added an `empty()` method, but I only use it in postive sense. I find `array.count() > 0` to be more readable than `!array.empty()`, because negations are hard.